Cannot execute arm64 release binaries

**Describe the bug**

Hello, Spectral folks. 👋 I receive the error "Error: Cannot find module '/snapshot/project/node_modules/tslib/tslib.js'" when executing the macos-arm64 binary at https://github.com/stoplightio/spectral/releases/tag/v6.10.0 . The full error is:

```
$ ./spectral-macos-arm64
pkg/prelude/bootstrap.js:1876
      throw error;
      ^

Error: Cannot find module '/snapshot/project/node_modules/tslib/tslib.js'
1) If you want to compile the package/file into executable, please pay attention to compilation warnings and specify a literal in 'require' call. 2) If you don't want to compile the package/file into executable and want to 'require' it from filesystem (likely plugin), specify an absolute path in 'require' call using process.cwd() or process.execPath.
    at createEsmNotFoundErr (node:internal/modules/cjs/loader:967:15)
    at finalizeEsmResolution (node:internal/modules/cjs/loader:960:15)
    at resolveExports (node:internal/modules/cjs/loader:488:14)
    at Module._findPath (node:internal/modules/cjs/loader:528:31)
    at Module._resolveFilename (node:internal/modules/cjs/loader:932:27)
    at Function._resolveFilename (pkg/prelude/bootstrap.js:1955:46)
    at Module._load (node:internal/modules/cjs/loader:787:27)
    at Module.require (node:internal/modules/cjs/loader:1012:19)
    at Module.require (pkg/prelude/bootstrap.js:1855:31)
    at require (node:internal/modules/cjs/helpers:102:18) {
  code: 'MODULE_NOT_FOUND',
  path: '/snapshot/project/node_modules/tslib/package.json',
  pkg: true
}

Node.js v18.5.0
```

The error happens on releases for 6.8.0 and 6.9.0 as well. I have also run into this error when running `spectral-alpine-arm64` from an Alpine container. I suspect the problem has been here since arm64 releases were published starting with the 6.8.0 release.

Also to note, I had to sign the macOS release via `codesign --sign - ./spectral-macos-arm64` to prevent my kernel from killing the program.

Thanks for looking and for the help!

**To Reproduce**

1. Download an arm64 release binary
2. Run `codesign` if you're on a mac
3. Execute the program
4. See error

**Expected behavior**

I expect to receive a usage statement since I'm not passing in any command arguments:

```
spectral <command>

Commands:
  spectral lint [documents..]  lint JSON/YAML documents from files or URLs

Options:
  --version  Show version number                                                                                                                                                                                     [boolean]
  --help     Show help
```

**Environment**

```
$ # My OS
$ sw_vers
ProductName:		macOS
ProductVersion:		13.5
BuildVersion:		22G74

$ # My kernel
$ uname -a
Darwin tome.local 22.6.0 Darwin Kernel Version 22.6.0: Wed Jul  5 22:22:52 PDT 2023; root:xnu-8796.141.3~6/RELEASE_ARM64_T8103 arm64

$ # My shell
$ zsh --version
zsh 5.9 (arm-apple-darwin21.3.0)
```

**Additional context**

I originally introduced arm64 builds in #2492. `pkg` execution worked fine locally on my macOS laptop. Looking at e9edd2efa0532d62de11a5cd7ebc95ec6bd139f4's [build](https://app.circleci.com/pipelines/github/stoplightio/spectral/9793/workflows/f8e6af01-eb87-4982-ba98-bcc33671a77f/jobs/37879), `pkg` executed with no error, but there are a large number of warnings in the build's [output](https://circleci.com/api/v1.1/project/github/stoplightio/spectral/37879/output/106/0?file=true&allocation-id=64c121f21b6c9428350a3d07-0-build%2FC71D872):

```
> Warning Entry 'main' not found in %1
  %1: /home/circleci/project/node_modules/@stoplight/spectral-formatters/package.json
  %2: /home/circleci/project/packages/cli/dist/services/output.js
> Warning Cannot find module '@stoplight/spectral-ruleset-bundler/presets/node' from '/home/circleci/project/packages/cli/dist/services/linter/utils'  in /home/circleci/project/packages/cli/dist/services/linter/utils/getRuleset.js
> Warning Cannot find module '@stoplight/spectral-ruleset-bundler/plugins/commonjs' from '/home/circleci/project/packages/cli/dist/services/linter/utils'  in /home/circleci/project/packages/cli/dist/services/linter/utils/getRuleset.js
> Warning Cannot find module '@stoplight/spectral-ruleset-bundler/plugins/stdin' from '/home/circleci/project/packages/cli/dist/services/linter/utils'  in /home/circleci/project/packages/cli/dist/services/linter/utils/getRuleset.js
> Warning Cannot find module '@stoplight/spectral-ruleset-bundler/plugins/builtins' from '/home/circleci/project/packages/cli/dist/services/linter/utils'  in /home/circleci/project/packages/cli/dist/services/linter/utils/getRuleset.js
> Warning Failed to make bytecode node18-arm64 for file /snapshot/project/packages/cli/dist/index.js
> Warning Failed to make bytecode node18-arm64 for file /snapshot/project/packages/cli/dist/commands/lint.js error (write EPIPE)
> Warning Failed to make bytecode node18-arm64 for file /snapshot/project/packages/cli/dist/errors/index.js
<snip>
```

There are 5,472 instances of these "Failed to make bytecode node18-arm64" warnings. I suspect this is the problem.

I was unable to reproduce the same build warnings following the same build steps in docker locally with CircleCI's `cimg/node:lts` container. The only differences between my setup and CircleCI's are the container runtime and that I don't have yarn dependencies cached. 

I can see if I can reproduce the same warnings and debug `pkg`, but if y'all have seen this before and know a quick fix then I'd be grateful. Thanks again for checking this out!

Any update on this? It seems https://github.com/stoplightio/spectral/releases/download/v6.11.0/spectral-linux-arm64  (`v6.11.0`) still returns the same error. I'm running spectral inside a Docker container of an image that curls the above package:
```
pkg/prelude/bootstrap.js:1876
      throw error;
      ^

Error: Cannot find module '/snapshot/project/node_modules/tslib/tslib.js'
1) If you want to compile the package/file into executable, please pay attention to compilation warnings and specify a literal in 'require' call. 2) If you don't want to compile the package/file into executable and want to 'require' it from filesystem (likely plugin), specify an absolute path in 'require' call using process.cwd() or process.execPath.
    at createEsmNotFoundErr (node:internal/modules/cjs/loader:967:15)
    at finalizeEsmResolution (node:internal/modules/cjs/loader:960:15)
    at resolveExports (node:internal/modules/cjs/loader:488:14)
    at Module._findPath (node:internal/modules/cjs/loader:528:31)
    at Module._resolveFilename (node:internal/modules/cjs/loader:932:27)
    at Function._resolveFilename (pkg/prelude/bootstrap.js:1955:46)
    at Module._load (node:internal/modules/cjs/loader:787:27)
    at Module.require (node:internal/modules/cjs/loader:1012:19)
    at Module.require (pkg/prelude/bootstrap.js:1855:31)
    at require (node:internal/modules/cjs/helpers:102:18) {
  code: 'MODULE_NOT_FOUND',
  path: '/snapshot/project/node_modules/tslib/package.json',
  pkg: true
}

Node.js v18.5.0
make: *** [lint_apis] Error 1
```

Please update npm in those latest packages and re release.

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -398,7 +398,6 @@ workflows:
           filters: *default-branch-filter
       - test-browser
       - lint
-      - build-nix-binaries
       - approve-release:
           type: approval
           filters: *default-branch-filter

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -376,6 +376,7 @@ workflows:
           filters: *default-branch-filter
       - test-browser
       - lint
+      - build-nix-binaries
       - approve-release:
           type: approval
           filters: *default-branch-filter

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -74,7 +74,7 @@ commands:
           name: Create the Spectral Binary
           environment:
             PKG_CACHE_PATH: *pkg-cache-path
-          command: yarn workspace @stoplight/spectral-cli pkg . --public --targets << parameters.targets >> --output binaries/spectral
+          command: yarn workspace @stoplight/spectral-cli pkg . --debug --public --targets << parameters.targets >> --output binaries/spectral
       - save_cache:
           name: Retain pkg cache
           key: *pkg-cache-key

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -22,9 +22,10 @@ defaults:
   node-versions: &node-versions
     - "12.22"
     - "14.21"
-    - &node-maintenance-lts "16.20"
-    - &node-active-lts "18.17"
-    - &node-current "20.5"
+    - "16.20"
+    - &node-maintenance-lts "18.19"
+    - &node-active-lts "20.10"
+    - &node-current "21.5"
   pkg-cache-path: &pkg-cache-path /tmp/pkg-node-binaries
   ldid-version: &ldid-version 2.1.5-procursus7 # https://github.com/ProcursusTeam/ldid
 
@@ -65,6 +66,12 @@ commands:
     parameters:
       targets:
         type: string
+      node-version:
+        type: enum
+        enum:
+          - *node-maintenance-lts
+          - *node-active-lts
+        default: *node-maintenance-lts
       bytecode:
         type: boolean
         default: true
@@ -72,18 +79,22 @@ commands:
       - restore_cache:
           name: Restore pkg cache
           keys:
-            - &pkg-cache-key pkg-cache-<< parameters.targets >>
+            - &pkg-cache-key pkg-cache-<< parameters.targets >>-<< parameters.node-version >>
       - run:
           name: Create the Spectral Binary
           environment:
             PKG_CACHE_PATH: *pkg-cache-path
           command: |
+            node_version=$(echo "<< parameters.node-version >>" | awk -F "." '{ print $1 }')
+            targets=$(echo "<< parameters.targets >>" | awk -v node_version="$node_version" -F "," '{for (i=1;i<=NF;i++) printf "node" node_version "-" $i "," }' | sed "s/,$/\n/")
             if [ "<< parameters.bytecode >>" = true ]
             then
-              yarn workspace @stoplight/spectral-cli pkg . --public --public-packages "*" --targets << parameters.targets >> --output binaries/spectral
+              yarn workspace @stoplight/spectral-cli pkg . --public --public-packages "*" --targets $targets --output binaries/spectral
             else
-              yarn workspace @stoplight/spectral-cli pkg . --no-bytecode --debug --public --public-packages "*" --targets << parameters.targets >> --output binaries/spectral
+              yarn workspace @stoplight/spectral-cli pkg . --no-bytecode --public --public-packages "*" --targets $targets --output binaries/spectral
             fi
+            
+
       - save_cache:
           name: Retain pkg cache
           key: *pkg-cache-key

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -65,6 +65,9 @@ commands:
     parameters:
       targets:
         type: string
+      bytecode:
+        type: boolean
+        default: true
     steps:
       - restore_cache:
           name: Restore pkg cache
@@ -74,12 +77,20 @@ commands:
           name: Create the Spectral Binary
           environment:
             PKG_CACHE_PATH: *pkg-cache-path
-          command: yarn workspace @stoplight/spectral-cli pkg . --debug --public --targets << parameters.targets >> --output binaries/spectral
+          command: |
+            if [ "<< parameters.bytecode >>" = true ]
+            then
+              yarn workspace @stoplight/spectral-cli pkg . --public --public-packages "*" --targets << parameters.targets >> --output binaries/spectral
+            else
+              yarn workspace @stoplight/spectral-cli pkg . --no-bytecode --debug --public --public-packages "*" --targets << parameters.targets >> --output binaries/spectral
+            fi
       - save_cache:
           name: Retain pkg cache
           key: *pkg-cache-key
           paths:
             - *pkg-cache-path
+      - store_artifacts:
+          path: /home/circleci/project/packages/cli/binaries
 
   install-and-build:
     description: >-
@@ -103,6 +114,8 @@ commands:
           name: Build ldid
           working_directory: /tmp
           command: |
+            sudo apt-get update -y
+            sudo apt-get -y install libplist-dev
             curl -L https://github.com/ProcursusTeam/ldid/archive/refs/tags/v<< parameters.version >>.tar.gz | tar -xz
             mv ldid-<< parameters.version >> ldid
             make -C ldid
@@ -254,15 +267,13 @@ jobs:
       - image: cimg/node:lts
     steps:
       - checkout
-      - run:
-          name: Configure binfmt
-          command: |
-            sudo apt-get update -y
-            sudo apt-get -y install binfmt-support qemu-user-static libplist-dev
       - install-ldid
       - install-and-build
+      # - build-binary:
+          # targets: linux-x64,macos-x64,alpine-x64
       - build-binary:
-          targets: linux-x64,linux-arm64,macos-x64,macos-arm64,alpine-x64,alpine-arm64
+          targets: macos-arm64,linux-arm64,alpine-arm64
+          bytecode: false
       - persist_to_workspace:
           root: ./packages/cli/
           paths:

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -6,7 +6,7 @@ orbs:
 executors:
   linux:
     docker:
-      - image: cimg/base:2021.07
+      - image: cimg/base:2024.01
 
 defaults:
   default-branch: &default-branch develop
@@ -26,6 +26,7 @@ defaults:
     - &node-active-lts "18.17"
     - &node-current "20.5"
   pkg-cache-path: &pkg-cache-path /tmp/pkg-node-binaries
+  ldid-version: &ldid-version 2.1.5-procursus7 # https://github.com/ProcursusTeam/ldid
 
 commands:
   compute-hashes:
@@ -65,11 +66,20 @@ commands:
       targets:
         type: string
     steps:
+      - restore_cache:
+          name: Restore pkg cache
+          keys:
+            - pkg-cache-{{ parameters.targets }}
       - run:
           name: Create the Spectral Binary
           environment:
             PKG_CACHE_PATH: *pkg-cache-path
           command: yarn workspace @stoplight/spectral-cli pkg . --public --targets << parameters.targets >> --output binaries/spectral
+      - save_cache:
+          name: Retain pkg cache
+          key: pkg-cache-{{ parameters.targets }}
+          paths:
+            - *pkg-cache-path
 
   install-and-build:
     description: >-
@@ -77,6 +87,34 @@ commands:
     steps:
       - cached-dependencies
       - build
+
+  install-ldid:
+    description: Install ldid
+    parameters:
+      version:
+        type: string
+        default: *ldid-version
+    steps:
+      - restore_cache:
+          name: Restore built ldid binary
+          keys:
+            - ldid-binary-<< parameters.version >>-{{ arch }}
+      - run:
+          name: Build ldid
+          working_directory: /tmp
+          command: |
+            curl -L https://github.com/ProcursusTeam/ldid/archive/refs/tags/v<< parameters.version >>.tar.gz | tar -xz
+            mv ldid-<< parameters.version >> ldid
+            make -C ldid
+      - save_cache:
+          name: Retain ldid binary
+          key: ldid-binary-<< parameters.version >>-{{ arch }}
+          paths:
+            - /tmp/ldid
+      - run:
+          name: Install ldid
+          working_directory: /tmp/ldid
+          command: make install
 
   lint-code:
     steps:
@@ -138,7 +176,6 @@ commands:
           key: test-harness-{{ checksum "/tmp/current-commit-sha" }}-<< parameters.os >>-<< parameters.node-version >>
           paths:
             - ./.cache/spectral-test-harness
-            - *pkg-cache-path
             - ./test-harness/tests/
           when: always
 
@@ -217,6 +254,12 @@ jobs:
       - image: cimg/node:lts
     steps:
       - checkout
+      - run:
+          name: Configure binfmt
+          command: |
+            sudo apt-get update -y
+            sudo apt-get -y install binfmt-support qemu-user-static libplist-dev
+      - install-ldid
       - install-and-build
       - build-binary:
           targets: linux-x64,linux-arm64,macos-x64,macos-arm64,alpine-x64,alpine-arm64

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -69,7 +69,7 @@ commands:
       - restore_cache:
           name: Restore pkg cache
           keys:
-            - pkg-cache-{{ parameters.targets }}
+            - &pkg-cache-key pkg-cache-<< parameters.targets >>
       - run:
           name: Create the Spectral Binary
           environment:
@@ -77,7 +77,7 @@ commands:
           command: yarn workspace @stoplight/spectral-cli pkg . --public --targets << parameters.targets >> --output binaries/spectral
       - save_cache:
           name: Retain pkg cache
-          key: pkg-cache-{{ parameters.targets }}
+          key: *pkg-cache-key
           paths:
             - *pkg-cache-path
 

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -22,10 +22,9 @@ defaults:
   node-versions: &node-versions
     - "12.22"
     - "14.21"
-    - "16.20"
-    - &node-maintenance-lts "18.19"
-    - &node-active-lts "20.10"
-    - &node-current "21.5"
+    - &node-maintenance-lts "16.20"
+    - &node-active-lts "18.17"
+    - &node-current "20.5"
   pkg-cache-path: &pkg-cache-path /tmp/pkg-node-binaries
   ldid-version: &ldid-version 2.1.5-procursus7 # https://github.com/ProcursusTeam/ldid
 
@@ -68,7 +67,7 @@ commands:
         type: string
       node-version:
         type: string
-        default: *node-maintenance-lts
+        default: *node-active-lts
       bytecode:
         type: boolean
         default: true
@@ -290,8 +289,6 @@ jobs:
           root: ./packages/cli/
           paths:
             - binaries
-      - store_artifacts:
-          path: /home/circleci/project/packages/cli/binaries
 
   build-windows-binary:
     executor:

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -124,9 +124,12 @@ commands:
           command: |
             sudo apt-get update -y
             sudo apt-get -y install libplist-dev
-            curl -L https://github.com/ProcursusTeam/ldid/archive/refs/tags/v<< parameters.version >>.tar.gz | tar -xz
-            mv ldid-<< parameters.version >> ldid
-            make -C ldid
+            if [ ! -d ldid ]
+            then
+              curl -L https://github.com/ProcursusTeam/ldid/archive/refs/tags/v<< parameters.version >>.tar.gz | tar -xz
+              mv ldid-<< parameters.version >> ldid
+              make -C ldid
+            fi
       - save_cache:
           name: Retain ldid binary
           key: ldid-binary-<< parameters.version >>-{{ arch }}

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -67,10 +67,7 @@ commands:
       targets:
         type: string
       node-version:
-        type: enum
-        enum:
-          - *node-maintenance-lts
-          - *node-active-lts
+        type: string
         default: *node-maintenance-lts
       bytecode:
         type: boolean
@@ -85,23 +82,23 @@ commands:
           environment:
             PKG_CACHE_PATH: *pkg-cache-path
           command: |
-            node_version=$(echo "<< parameters.node-version >>" | awk -F "." '{ print $1 }')
-            targets=$(echo "<< parameters.targets >>" | awk -v node_version="$node_version" -F "," '{for (i=1;i<=NF;i++) printf "node" node_version "-" $i "," }' | sed "s/,$/\n/")
+            NODE_VERSION=$(echo "<< parameters.node-version >>" | awk -F "." '{ print $1 }')
+            TARGETS=$(echo "<< parameters.targets >>" | awk -v node_version="$NODE_VERSION" -F "," '{for (i=1;i<=NF;i++) printf "node" node_version "-" $i "," }' | sed "s/,$/\n/")
+            export ARCH=$(echo << parameters.targets >> | awk 'match($0, /-(arm64|x64)/){ print substr($0, RSTART+1, RLENGTH-1) }')
+
             if [ "<< parameters.bytecode >>" = true ]
             then
-              yarn workspace @stoplight/spectral-cli pkg . --public --public-packages "*" --targets $targets --output binaries/spectral
+              yarn workspace @stoplight/spectral-cli pkg . --public --public-packages "*" --targets $TARGETS --output binaries/spectral
             else
-              yarn workspace @stoplight/spectral-cli pkg . --no-bytecode --public --public-packages "*" --targets $targets --output binaries/spectral
+              yarn workspace @stoplight/spectral-cli pkg . --no-bytecode --public --public-packages "*" --targets $TARGETS --output binaries/spectral
             fi
             
-
+            (cd packages/cli/binaries && ls . | sed -n -E 's/^(spectral)-(alpine|linux|macos)$/mv "\1-\2" "\1-\2"-$ARCH/p' | sh)
       - save_cache:
           name: Retain pkg cache
           key: *pkg-cache-key
           paths:
             - *pkg-cache-path
-      - store_artifacts:
-          path: /home/circleci/project/packages/cli/binaries
 
   install-and-build:
     description: >-
@@ -192,6 +189,7 @@ commands:
             - test-harness-{{ checksum "/tmp/previous-commit-sha" }}-<< parameters.os >>-<< parameters.node-version >>
       - build-binary:
           targets: << parameters.os >>
+          node-version: << parameters.node-version >>
       - run:
           name: Run harness tests
           command: yarn test.harness --maxWorkers=<< parameters.max-workers >>
@@ -280,8 +278,8 @@ jobs:
       - checkout
       - install-ldid
       - install-and-build
-      # - build-binary:
-          # targets: linux-x64,macos-x64,alpine-x64
+      - build-binary:
+          targets: linux-x64,macos-x64,alpine-x64
       - build-binary:
           targets: macos-arm64,linux-arm64,alpine-arm64
           bytecode: false
@@ -289,6 +287,8 @@ jobs:
           root: ./packages/cli/
           paths:
             - binaries
+      - store_artifacts:
+          path: /home/circleci/project/packages/cli/binaries
 
   build-windows-binary:
     executor:

--- a/packages/cli/package.json
+++ b/packages/cli/package.json
@@ -58,9 +58,9 @@
     "@types/es-aggregate-error": "^1.0.2",
     "@types/xml2js": "^0.4.9",
     "@types/yargs": "^17.0.8",
+    "@yao-pkg/pkg": "^5.11.1",
     "es-aggregate-error": "^1.0.7",
     "nock": "^13.1.3",
-    "pkg": "^5.8.0",
     "xml2js": "^0.5.0"
   },
   "pkg": {

